src/server/status/lib/metrics.js

```diff
@@ -56,53 +59,53 @@ export class Metrics {
 
     const metrics = {
       last_updated: timestamp,
-      collection_interval_in_millis: this.config.get('ops.interval'),
-      uptime_in_millis: event.process.uptime_ms, // TODO: deprecate this field, data should only have process.uptime_ms
+      collection_interval_in_millis: this.config.get('ops.interval')
     };
 
     return merge(metrics, event, cgroup);
   }
 
   captureEvent(hapiEvent) {
+    const heapStats = v8.getHeapStatistics();
     const port = this.config.get('server.port');
-
     const avgInMillis = get(hapiEvent, ['responseTimes', port, 'avg']); // sadly, it's possible for this to be NaN
     const maxInMillis = get(hapiEvent, ['responseTimes', port, 'max']);
 
     return {
       process: {
-        mem: {
-          // https://nodejs.org/docs/latest-v8.x/api/process.html#process_process_memoryusage
-          heap_max_in_bytes: get(hapiEvent, 'psmem.heapTotal'),
-          heap_used_in_bytes: get(hapiEvent, 'psmem.heapUsed'),
+        memory: {
+          heap: {
+            // https://nodejs.org/docs/latest-v8.x/api/process.html#process_process_memoryusage
+            total_in_bytes: get(hapiEvent, 'psmem.heapTotal'),
+            used_in_bytes: get(hapiEvent, 'psmem.heapUsed'),
+            size_limit: heapStats.heap_size_limit
+          },
           resident_set_size_in_bytes: get(hapiEvent, 'psmem.rss'),
-          external_in_bytes: get(hapiEvent, 'psmem.external')
         },
+        event_loop_delay: get(hapiEvent, 'psdelay'), // TODO: should be event_loop_delay_ms
```

Comment:
I think we should either remove this TODO comment or add more context to it. I'm worried that someone might see this comment and just make the change without realizing the implications it causes for downstream consumers (metricbeat, monitoring UI), etc.